### PR TITLE
Allow opt-in go cli scanning

### PR DESCRIPTION
**Pull request by tevoinea**
Using go cli to scan is more accurate than parsing go.sum files, but go projects aren't always restored in a workflow. This results in go restoring packages in a workflow where it's not required. By allowing an opt-in we give users the power to make the right choice for their specific situations.

We still fall back to manual parsing if go cli fails.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
             "request": "launch",
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
-            "program": "${workspaceFolder}/src/Microsoft.ComponentDetection/bin/Debug/net6.0/Microsoft.ComponentDetection.dll",
+            "program": "${workspaceFolder}/src/Microsoft.ComponentDetection/bin/Debug/netcoreapp3.1/Microsoft.ComponentDetection.dll",
```
Comment (tevoinea) [inline .vscode/launch.json:13]: Tiny thing we missed moving back to netcore 3.1

File: src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs [hunk 2]
```diff
@@ -34,35 +38,62 @@ public class GoComponentDetector : FileComponentDetector
...
+                if (IsGoCliManuallyEnabled())
+                {
+                    Logger.LogInfo("Go cli scan was manually enabled");
+                    wasGoCliScanSuccessful = await UseGoCliToScan(file.Location, singleFileComponentRecorder);
+                }
+            }
+            catch
+            {
+                Logger.LogInfo("Failed to detect components using go cli.");
+            }
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs:64]: Can you log the exception here

File: src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs [hunk 3]
```diff
@@ -223,5 +254,10 @@ private bool TryCreateGoComponentFromRelationshipPart(string relationship, out G
             goComponent = new GoComponent(componentParts[0], componentParts[1]);
             return true;
         }
+
+        private bool IsGoCliManuallyEnabled()
+        {
+            return EnvVarService.DoesEnvironmentVariableExist("EnableGoCliScan");
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs:260]: Can you create a constants class for environment variables? If we're going to create a lot of these, it'll be easier to keep track of.

**Review by JamieMagee — APPROVED** [review]
Couple of small things, but nothing blocking.

**Pull request merged**